CHANGELOG.md

```diff
@@ -1,6 +1,115 @@
+- The `cache_config.yml` has now also been **deprecated** and merged into the `config.json`, as part of the profile options.
+  This merge will be handled automatically, upon first load of the `config.json` using the new AiiDA version.
+
+In-line with these changes, the `verdi config` command has been refactored into separate commands, including `verdi config list`, `verdi config set`, `verdi config unset` and `verdi config caching`.
+
+See the [Configuring profile options](https://aiida.readthedocs.io/projects/aiida-core/en/v1.6.0/howto/installation.html#configuring-profile-options) and [Configuring caching](https://aiida.readthedocs.io/projects/aiida-core/en/v1.6.0/howto/run_codes.html#how-to-save-compute-time-with-caching) how-tos for more details.
+
+### Command-line additions and improvements 👌
+
+In addition to `verdi config`, numerous other new commands and options have been added to `verdi`:
+
+- **Deprecate** `verdi export` and `verdi import` commands and merge into `verdi archive` ([#4710](https://github.com/aiidateam/aiida-core/pull/4710))
+- Add `verdi group delete --delete-nodes`, to also delete the nodes in a group during its removal ([#4578](https://github.com/aiidateam/aiida-core/pull/4578)).
+- Improve `verdi group remove-nodes` command, to warn when requested nodes are not in the specified group ([#4728](https://github.com/aiidateam/aiida-core/pull/4728)).
+- Add `verdi database summary` ([#4737](https://github.com/aiidateam/aiida-core/pull/4737)):
+  This prints a summary of the count of each entity and (optionally) the list of unique identifiers for some entities.
+- Improve `verdi process play` performance, by only querying for active processes with the `--all` flag ([#4671](https://github.com/aiidateam/aiida-core/pull/4671))
+- Add the `verdi database version` command ([#4613](https://github.com/aiidateam/aiida-core/pull/4613)):
+  This shows the schema generation and version of the database of the given profile, useful mostly for developers when debugging.
+- Improve `verdi node delete` performance ([#4575](https://github.com/aiidateam/aiida-core/pull/4575)):
+  The logic has been re-written to greatly reduce the time to delete large amounts of nodes.
+- Fix `verdi quicksetup --non-interactive`, to ensure it does not include any user prompts ([#4573](https://github.com/aiidateam/aiida-core/pull/4573))
+- Fix `verdi --version` when used in editable mode ([#4576](https://github.com/aiidateam/aiida-core/pull/4576))
+
+### API additions and improvements 👌
+
+Base `Node` `__eq__` and `__hash__` methods are now based on a node's UUID ([#4753](https://github.com/aiidateam/aiida-core/pull/4753)).
+For example, loading the same node twice will always resolve as equivalent: `load_node(1) == load_node(1)`, and nodes can be correctly added to sets: `len({load_node(1), load_node(1)}) == 1`.
```

Comment:
hence why I said the commit message should have been better 😆 

> We just set the `__hash__` for the nodes that are not mocking python intrinsics

Oh thats a bit crap: Note, from ticket [1549](https://bugs.python.org/issue1549):

> This was done intentionally -- if you define a comparison without defining a hash, the default hash will not match your comparison, and your objects will misbehave when used as dictionary keys. (Guido van Rossum)

Reply:
Mmm maybe. Anyways, just for context: we were not trying to go against the intentionality of the feature, this was already behaving like that before because of the special definition of `__eq__` in the python type mocking node types. In the modification we did for the  `Node` base class we set the `__hash__` to behave the same as the `__eq__` for the fallback. We didn't do anything to the subclass because of the [unresolved discussion](https://github.com/aiidateam/aiida-core/issues/1917) of how that should behave so we just let it as it was.

Reply:
done